### ISSUE TITLE
Make Piecewise Functions and If-Expressions Core

### ISSUE THREAD

**Issue by flannery-denny** [bootstrapworld/curriculum#590]
This issue combines (#527), (#561),  and (#587)
Which pathway should we do this work in? **Separate if-expression branch out of Fall2022**
- [x] Should we call it if expressions or piecewise functions? **Piecewise Functions and If-Expressions**
- [x] Decide which if-else format to use **if-else in pyret with addendum that acknowledges the possibility of ask - in wescheme use cond**
- [x] Write a core programming lesson that introduces branching 
- Focuses on Mood generator 
- Alice's Restaurant becomes optional. 
- Old Design Recipe called red shape?
#664
- [x] Review Algebra less for implementing this in the game 
- [x] write separate lesson for Data Science on introducing this in image scatter plots (made this a separate issue (#634) .
#663
- [x] remove vertical line test context from if-expressions - add it to the lesson Emmanuel wants to build on functions that we saved other similar core fragments for?
- [x] WeSchemify Mood generator
- [x] Make Wescheme slides

**Comment by flannery-denny** [comment]
piecewise function ideas: 
- ticket prices by age: student & senior discounts
- calculate pet anesthesia dosage for Acepromazine
> Dog: 0.005 to 0.03 mg/lb  max total dosage regardless of weight 2 to 3mg
Cat: 0.03 to 0.05 mg/lb

**Comment by flannery-denny** [comment]
@schanzer Does the formatting of these Mood Generator File work for you:

- [Mood Generator WeScheme](https://www.wescheme.org/view?publicId=BTm0e7RDk6)
- [Mood Generator Pyret - IF-ELSE](https://code.pyret.org/editor#share=1chY8wU4UcNB8R9dRDOVhYkec0mK1Nipg&v=1904b2c)
- [Mood Generator Pyret - ASK](https://code.pyret.org/editor#share=1hAtb1ZyhjVHWMWiOb24UhV5XyBU_vK_m&v=1904b2c)

**Comment by flannery-denny** [comment]
@schanzer Could you please share the `red shape` file so that I can take a look at it?

**Comment by schanzer** [comment]
@flannery-denny 
The EXAMPLEs in WeScheme are indented for some reason? Also, I'm not sold on writing on the part of the recipe in all-caps

`red-shape` : this one goes back more than a decade, well before Pyret! Here's the code:
```
; red-shape :: String -> Image
; consumes a shape name, and produces a red version of that shape

(EXAMPLE (red-shape "circle") (circle 20 "solid" "red"))
(EXAMPLE (red-shape "triangle") (triangle 20 "solid" "red"))
(EXAMPLE (red-shape "rectangle") (rectangle 50 20 "solid" "red"))
(EXAMPLE (red-shape "star") (star 20 "solid" "red"))

(define (red-shape shape)  
  (cond 
    [(string=? shape "circle") (circle 20 "solid" "red")]
    [(string=? shape "triangle")   (triangle 20 "solid" "red")]
    [(string=? shape "rectangle") (rectangle 50 20 "solid" "red")]
    [(string=? shape "star")  (star 20 "solid" "red")]
    [else (text "Unknown shape name" 20 "red")]))
```

**Comment by flannery-denny** [comment]
@schanzer I'm happy to follow whatever template re parts of the recipe that
you would like in starter files. I got those caps from looking at our
starter files. I would like our materials to be consistent. Please identify
a starter file whose formatting is your preference and I will make sure the
rest match.

**Comment by flannery-denny** [comment]
@schanzer 

1. Do you think this labeling is helpful to students or should be removed?

> ; CONTRACT & PURPOSE
; mood :: String -> String
; consumes a mood, and produces the emoji for that mood

> ; EXAMPLES
(EXAMPLE (mood "happy") "😀")
(EXAMPLE (mood "sad")   "😥")
(EXAMPLE (mood "angry") "😡")
(EXAMPLE (mood "sick")  "🤮")

> ; FUNCTION DEFINITION
(define (mood feeling)  
  (cond 
    [(string=? feeling "happy") "😀"]
    [(string=? feeling "sad")   "😥"]
    [(string=? feeling "angry") "😡"]
    [(string=? feeling "sick")  "🤮"]
    [else "❓"]))

2. Do you think  # CONTRACT, # PURPOSE, #EXAMPLE and #DEFINiTION should be removed from all other starter files?

**Comment by schanzer** [comment]
@flannery-denny

1. I don't think they're terribly useful in this lesson. 
2. See my previous comment.

**Comment by flannery-denny** [comment]
@schanzer - if you find yourself in the headspace where reading a lesson plan would be fun, the first draft is ready. If you don't get to it, no worries, I'll likely delete this comment and come back to it with fresh eyes.

Questions:

- Is it important to discuss the Design Recipe?  For me, it's a holdover from other materials. If you're ok with it not being part of this lesson, I'd remove it. If you want it to stay, I might decide to flesh it out more.
- How do you feel about offering two starter files with scaffolding in the way that I did?

[index.pdf](https://github.com/bootstrapworld/curriculum/files/7333654/index.pdf)
[MoodGenerator-intro.pdf](https://github.com/bootstrapworld/curriculum/files/7333660/MoodGenerator-intro.pdf)
[RedShape-intro.pdf](https://github.com/bootstrapworld/curriculum/files/7333662/RedShape-intro.pdf)

If you want to see the html version checkout out to the `if-expressions` branch and build the `__sample` pathway.

**Comment by flannery-denny** [comment]
@schanzer - When If Expressions and Image scatterplots were one lesson, they came before scatterplots. Do you still want it to be this way? Or might it make sense to think of Image Scatterplots as a lesson that follows scatterplots now?

**Comment by flannery-denny** [comment]
@schanzer are you sure the line needs to break where it does in the red-shape-dr page? I think these examples would be probably easier for students to circle and label if they fit on one line.
![Screen Shot 2021-11-04 at 6 08 02 PM](https://user-images.githubusercontent.com/66691020/140432594-94564ae2-58c4-4166-a045-11df4ad6e30d.png)

**Comment by schanzer** [comment]
The only way to make examples fit on one line is to make them shorter.

I'll make the fall2022 folder.

Language tables only exist in the context of a pathway - if you don't specify a pathway, there's no table. The other stuff should be on @ds26gte 's list

**Comment by flannery-denny** [comment]
@schanzer I made all of the examples shorter. It really looks like the line break is unnecessary. Are you sure there's no adjustment that can be made?
[redshape-dr.pdf](https://github.com/bootstrapworld/curriculum/files/7487178/redshape-dr.pdf)

**Comment by schanzer** [comment]
Finally had a chance to build and investigate. WHOA. Something is seriously wrong here. 

@ds26gte are you sure `if-expr` is up-to-date with  master? The Design Recipe CSS Flannery is seeing is ancient, and from the brittle pre-`display: flex` days. On master, design recipes do the right thing. But on her branch, it's still using ancient CSS.

**Comment by flannery-denny** [comment]
@schanzer 

1. I'm noticing that the last section of this lesson doesn't have a synthesize section and that the final pedagogy note is more or less a synthesize. Do we want the lesson to conform to the structure? or is it ok as is?
2. I'm noticing that the checkboxes at the top of this issue have a few ideas that are for the future and don't belong on the December 1 deadline. Should we make separate non-urgent issues for them so we don't lose track and close this issue once slide decks and review of the Player animation lesson are done?

**Comment by schanzer** [comment]
@flannery-denny 
1. I think we should aim to conform with the structure as much as possible.
2. Yes, let's make a new issue and close this one.

**Comment by flannery-denny** [comment]
@schanzer lesson plan got some updates and pyret slides are done.  please make whatever edits you'd like to both and let me know when it's ready for weschemification.

**Comment by schanzer** [comment]
@flannery-denny I made some small wording changes to the lesson and slides. Two things to bring to your attention:

1. A larger change: I moved the synthesis questions that pertain to the Design Recipe into the synthesis section for the Design Recipe.
2. A question: you mention a student with "purple eyes". Where does this come from? What does it mean?

Aside from these two issues, you're all clear for WeScheme.

**Comment by flannery-denny** [comment]
@schanzer I didn't write that box for Algebra 2 and because it seemed like a stand alone thing I haven't read it in forever. The piecewise function used to be about eye color, but I changed it to seasons... which I think works much better for the diversity of populations we are serving, but doesn't work as well for this break out box. I took it out, but wonder what you'd think about deleting the box altogether.
